api/src/main/java/org/pmiops/workbench/actionaudit/targetproperties/ModelBackedTargetProperty.kt

```diff
@@ -0,0 +1,9 @@
+package org.pmiops.workbench.actionaudit.targetproperties
+
+// This interface is a contract to allow any model type T to support
+// arbitrary String properties without any changes to T. Intended
+// to be implemented by enum classes.
+interface ModelBackedTargetProperty<T> : SimpleTargetProperty {
+    override val propertyName: String
```

Comment:
Does this string value have to exactly match a property name? Or is this just for display purposes in the audit log? Would be good to specify here

Reply:
This string is as close as we get to a formal schema for the audit log, so these values shouldn't change. They may often match field names on the model object, but by design I'm trying to avoid relying on those staying the same forever. Most of the schema is supposed to use user-friendly names and concepts, without being coupled to our DB (with the exception of ID columns).

Reply:
Please capture this in the interface comment here